packages/shared/prisma/schema.prisma

```diff
@@ -1214,3 +1218,108 @@ model TableViewPreset {
 }
+
+model Action {
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now()) @map("created_at")
+  updatedAt DateTime @default(now()) @updatedAt @map("updated_at")
+
+  projectId String  @map("project_id")
+  project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+  type ActionType
+
+  // Configuration specific to each action type
+  config Json // Structured JSON for different action types
+  // For WEBHOOK: { version: "1.0", url: "...", method: "POST", headers: {...}, secretId: "..." }
+  // For ANNOTATION_QUEUE: { version: "1.0", queueId: "..." }
+
+  // Relations
+  triggers        Automation[] // Changed to use the join table
+  ActionExecution ActionExecution[]
+
+  @@index([projectId])
+  @@map("actions")
+}
+
+model Trigger {
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now()) @map("created_at")
+  updatedAt DateTime @default(now()) @updatedAt @map("updated_at")
+
+  projectId String  @map("project_id")
+  project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+  // When should this trigger fire
+  eventSource  String // trace, prompt, etc.
+  eventActions String[] // created, updated, deleted
+  filter       Json? // Filter conditions (format: { field: "name", operator: "equals", value: "my_trace" })
+
+  // What action to perform when triggered - removed direct relation and added many-to-many
+  actions Automation[]
+
+  // Additional attributes
+  status JobConfigState @default(ACTIVE) @map("status")
+
+  // Link to executions
+  ActionExecution ActionExecution[]
+
+  @@index([projectId])
+  @@map("triggers")
+}
+
+model Automation {
+  id        String   @id @default(cuid())
+  name      String   @map("name")
+  trigger   Trigger  @relation(fields: [triggerId], references: [id], onDelete: Cascade)
+  triggerId String   @map("trigger_id")
+  action    Action   @relation(fields: [actionId], references: [id], onDelete: Cascade)
+  actionId  String   @map("action_id")
+  createdAt DateTime @default(now()) @map("created_at")
+  projectId String   @map("project_id")
+  project   Project  @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+  @@index([projectId, actionId, triggerId])
+  @@index([projectId, name])
+  @@map("automations")
+}
+
+enum ActionType {
+  WEBHOOK
+  // More action types can be added as needed
+}
+
+enum ActionExecutionStatus {
+  COMPLETED
+  ERROR
+  PENDING
+  CANCELLED
+}
+
+model ActionExecution {
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now()) @map("created_at")
+  updatedAt DateTime @default(now()) @updatedAt @map("updated_at")
+  sourceId  String   @map("source_id")
+
+  triggerId String  @map("trigger_id")
+  trigger   Trigger @relation(fields: [triggerId], references: [id], onDelete: Cascade)
+
+  actionId String @map("action_id")
+  action   Action @relation(fields: [actionId], references: [id], onDelete: Cascade)
```

Comment:
Just putting it out there, unclear whether it is better: We can consider referencing the Automation here. Otherwise, we may be in an "underdefined" situation in case someone creates two Automations with the same triggers and actions.
Also, an AutomationExecution feels more natural in that regard.

Alternatively, we can skip the trigger here and only bind it to actions.